dash-pipeline/bmv2/dash_pipeline.p4

```diff
@@ -172,99 +201,59 @@ control dash_ingress(
-        set_meter_attrs(meta, meter_class_or, meter_class_and);
+// direction lookup, eni lookup, etc
+control dash_lookup_stage(
+      inout headers_t hdr
+    , inout metadata_t meta
+    )
+{
+    action deny() {
+        meta.dropped = true;
     }
 
-    action tunnel_decap_pa_validate(inout headers_t hdr,
-                                    inout metadata_t meta,
-                                    @SaiVal[type="sai_object_id_t"] bit<16> src_vnet_id,
-                                    bit<32> meter_class_or,
-                                    @SaiVal[default_value="4294967295"] bit<32> meter_class_and) {
-        meta.vnet_id = src_vnet_id;
-        set_meter_attrs(meta, meter_class_or, meter_class_and);
+    action accept() {
     }
 
-    @SaiTable[name = "pa_validation", api = "dash_pa_validation"]
-    table pa_validation {
+    @SaiTable[name = "vip", api = "dash_vip"]
+    table vip {
         key = {
-            meta.vnet_id: exact @SaiVal[type="sai_object_id_t"];
-            hdr.u0_ipv4.src_addr : exact @SaiVal[name = "sip", type="sai_ip_address_t"];
+            hdr.u0_ipv4.dst_addr : exact @SaiVal[name = "VIP", type="sai_ip_address_t"];
         }
 
         actions = {
-            permit;
-            @defaultonly deny;
-        }
-
-        const default_action = deny;
-    }
-
-    @SaiTable[name = "inbound_routing", api = "dash_inbound_routing"]
-    table inbound_routing {
-        key = {
-            meta.eni_id: exact @SaiVal[type="sai_object_id_t"];
-            hdr.u0_vxlan.vni : exact @SaiVal[name = "VNI"];
-            hdr.u0_ipv4.src_addr : ternary @SaiVal[name = "sip", type="sai_ip_address_t"];
-        }
-        actions = {
-            tunnel_decap(hdr, meta);
-            tunnel_decap_pa_validate(hdr, meta);
-            vxlan_decap;                // Deprecated, but cannot be removed until SWSS is updated.
-            vxlan_decap_pa_validate;    // Deprecated, but cannot be removed until SWSS is updated.
+            accept;
             @defaultonly deny;
         }
 
         const default_action = deny;
     }
 
-    action set_acl_group_attrs(@SaiVal[type="sai_ip_addr_family_t", isresourcetype="true"] bit<32> ip_addr_family) {
-        if (ip_addr_family == 0) /* SAI_IP_ADDR_FAMILY_IPV4 */ {
-            if (meta.is_overlay_ip_v6 == 1) {
-                meta.dropped = true;
-            }
-        } else {
-            if (meta.is_overlay_ip_v6 == 0) {
-                meta.dropped = true;
-            }
-        }
+    action set_appliance(EthernetAddress neighbor_mac,
+                         EthernetAddress mac) {
+        meta.encap_data.underlay_dmac = neighbor_mac;
+        meta.encap_data.underlay_smac = mac;
     }
 
-    @SaiTable[name = "dash_acl_group", api = "dash_acl", isobject="true"]
-    table acl_group {
+    /* This table API should be implemented manually using underlay SAI */
+    @SaiTable[ignored = "true"]
+    table appliance {
```

Comment:
need to sync to latest code. this is changed now.

Reply:
fixed